testcases/OpTestRTCdriver.py

```diff
@@ -34,69 +34,28 @@
-    #  @param i_bmcUser The userid to log into the BMC with
-    #  @param i_bmcPasswd The password of the userid to log into the BMC with
-    #  @param i_bmcUserIpmi The userid to issue the BMC IPMI commands with
-    #  @param i_bmcPasswdIpmi The password of BMC IPMI userid
-    #  @param i_ffdcDir Optional param to indicate where to write FFDC
-    #
-    # "Only required for inband tests" else Default = None
-    # @param i_hostIP The IP address of the HOST
-    # @param i_hostuser The userid to log into the HOST
-    # @param i_hostPasswd The password of the userid to log into the HOST with
-    #
-    def __init__(self, i_bmcIP, i_bmcUser, i_bmcPasswd,
-                 i_bmcUserIpmi, i_bmcPasswdIpmi, i_ffdcDir=None, i_hostip=None,
-                 i_hostuser=None, i_hostPasswd=None):
-        self.cv_BMC = OpTestBMC(i_bmcIP, i_bmcUser, i_bmcPasswd, i_ffdcDir)
-        self.cv_IPMI = OpTestIPMI(i_bmcIP, i_bmcUserIpmi, i_bmcPasswdIpmi,
-                                  i_ffdcDir)
-        self.cv_HOST = OpTestHost(i_hostip, i_hostuser, i_hostPasswd,i_bmcIP)
-        self.cv_SYSTEM = OpTestSystem(i_bmcIP, i_bmcUser, i_bmcPasswd,
-                 i_bmcUserIpmi, i_bmcPasswdIpmi, i_ffdcDir, i_hostip,
-                 i_hostuser, i_hostPasswd)
+class FullRTC(unittest.TestCase):
+    def setUp(self):
+        conf = OpTestConfiguration.conf
+        self.cv_HOST = conf.host()
+        self.cv_IPMI = conf.ipmi()
+        self.cv_SYSTEM = conf.system()
         self.util = OpTestUtil()
 
-    ##
-    # @brief This function will cover following test steps
-    #        1. Getting host information(OS and Kernel info)
-    #        2. Loading rtc_opal module based on config option
-    #        3. Testing the rtc driver functions
-    #                Display the current time,
-    #                set the Hardware Clock to a specified time,
-    #                set the Hardware Clock from the System Time, or
-    #                set the System Time from the Hardware Clock
-    #                keep the Hardware clock in UTC or local time format
-    #                Hardware clock compare, predict and adjust functions
-    #                Hardware clock debug and test modes
-    #                Reading the Hardware clock from special file instead of default
-    #        4. After executing above each function reading the Hardware clock in b/w functions.
-    #
-    # @return BMC_CONST.FW_SUCCESS or raise OpTestError
-    #
-    def test_RTC_driver(self):
-        self.cv_SYSTEM.sys_bmc_power_on_validate_host()
-
-        # Get OS level
-        l_oslevel = self.cv_HOST.host_get_OS_Level()
-
-        # Get Kernel Version
-        l_kernel = self.cv_HOST.host_get_kernel_version()
+    def test_init(self):
+        # TODO: run in skiroot too
+        self.cv_SYSTEM.goto_state(OpSystemState.OS)
 
         # Get hwclock version
         l_hwclock = self.cv_HOST.host_run_command("hwclock -V;echo $?")
```

Comment:
I'd prefer we error out. If the OS running doesn't have everything we need to run the test, I think we should fail and make somebody fix that OS install rather than risk not running the test